src/sys/pthread.rs

```
@@ -36,3 +38,29 @@ pub fn pthread_kill<T: Into<Option<Signal>>>(thread: Pthread, signal: T) -> Resu
     let res = unsafe { libc::pthread_kill(thread, sig) };
     Errno::result(res).map(drop)
 }
+
+/// Obtain the name of the thread
+///
+/// On linux the name cannot exceed 16 length including null terminator
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+pub fn pthread_getname_np(thread: Pthread) -> Result<CString> {
+    let mut name: Vec<u8> = Vec::with_capacity(16);
+    unsafe { name.set_len(16); }
+    unsafe { libc::pthread_getname_np(thread, name.as_mut_ptr() as _, name.len()) };
+    let zi = name.iter().position(|x| *x == b'\0').unwrap();
+    name.truncate(zi);
+    Ok(unsafe { CString::from_vec_unchecked(name) })
+}
+
+/// Set the name of the thread
+#[cfg(all(target_os = "linux", not(target_env = "musl")))]
+pub fn pthread_setname_np<T: Into<Vec<u8>>>(thread: Pthread, name: T) -> Result<()> {
+    match CString::new(name.into()) {
+        Ok(cname) => {
+            let nameptr = cname.as_ptr();
+            let res = unsafe { libc::pthread_setname_np(thread, nameptr) };
+            Errno::result(res).map(drop)
+        },
+        Err(_) => Ok(()),
```

Don't blindly ignore errors.  You should return an error in this case.